test.py

```diff
@@ -1,131 +1,194 @@
+        17,
+        19,
+        23,
+        29,
+        31,
+        37,
+        41,
+        43,
+        47,
+        53,
+        59,
+        61,
+        67,
+        71,
+        73,
+        79,
+        83,
+        89,
+        97,
+    ]
+    n_Q_power2 = [2, 4, 8, 16, 32, 64]
+    n_Q_values = n_Q_prime + n_Q_power2 + [n_T]
+    n_Q_values = sorted(n_Q for n_Q in set(n_Q_values) if n_Q <= n_T)
+
+    modules = utils.import_sdp_mods()
+    for n_Q in n_Q_values:
+        Q = np.random.rand(n_Q)
+        T = np.random.rand(n_T)
+        ref = naive_sliding_dot_product(Q, T)
+        for mod in modules:
+            try:
+                comp = mod.sliding_dot_product(Q, T)
+                npt.assert_allclose(comp, ref)
+            except Exception as e:  # pragma: no cover
+                msg = f"Error in {mod.__name__}, with n_Q={n_Q} and n_T={n_T}"
+                print(msg)
+                raise e
+
+    return
+
+
+def test_sdp_power2():
+    # test for case 5. len(T) is power of 2
+    pmin = 3
+    pmax = 13
+
+    modules = utils.import_sdp_mods()
+    for mod in modules:
+        try:
+            for q in range(pmin, pmax + 1):
+                n_Q = 2**q
+                for p in range(q, pmax + 1):
+                    n_T = 2**p
+                    Q = np.random.rand(n_Q)
+                    T = np.random.rand(n_T)
 
+                    ref = naive_sliding_dot_product(Q, T)
+                    comp = mod.sliding_dot_product(Q, T)
+                    npt.assert_allclose(comp, ref)
 
-def func_exists(mod_path, func_name):
-    try:
-        with open(mod_path, "r") as file:
-            module_content = file.read()
-    except FileNotFoundError:
-        return False  # Module file not found
+        except Exception as e:  # pragma: no cover
+            msg = f"Error in {mod.__name__}, with q={q} and p={p}"
+            print(msg)
+            raise e
 
-    try:
-        tree = ast.parse(module_content)
-    except SyntaxError:
-        return False  # Syntax error in module
+    return
 
-    for node in ast.walk(tree):
-        if isinstance(node, ast.FunctionDef) and node.name == func_name:
-            return True
-    return False
 
+def test_setup():
+    Q = np.random.rand(3)
+    T = np.random.rand(10)
 
-def import_sdp_mods(include=None, ignore=None):
-    mods = []
     for m in sorted(list(pkgutil.iter_modules(sdp.__path__))):
-        mod_path = f"sdp/{m[1]}.py"
-        if (
-            include is not None
-            and len(include)
-            and not any(mod in mod_path for mod in include)
-        ):
-            continue
-        if (
-            ignore is not None
-            and len(ignore)
-            and any(mod in mod_path for mod in ignore)
-        ):
-            continue
-
-        if (
-            "sdp" in m[1]
-            and func_exists(mod_path, "sliding_dot_product")
-            and func_exists(mod_path, "setup")
-        ):
+        if m[1].endswith("_sdp"):
```

Comment:
@NimaSarajpoor 
The test function now checks all modules that end with `_sdp`. To make it consistent with what we have in `utils.import_sdp_mods`, we should change this condition to: `if sdp in m[1]`.